### DISCUSSION TITLE
Use CIP-22 as a way to identify SPOs when registering keys

### DISCUSSION THREAD

**Discussion by ghost** [input-output-hk/mithril#507] [Ideas]
Following a discussion on Discord on the acceptability of signing-based registration process, somebody pointed me at https://cips.cardano.org/cips/cip22/ which could be a good fit to remove the need for SPOs to run Mithril signer alongside block producers.

**Comment by reqlez** [comment]
If it can be implemented, i think it will be nice, since SPOs are already used to this process for verification, so it will be easier to get people on-boarded.

**Reply by ghost** [reply]
We discussed this topic this afternoon with @jpraynaud and @ghubertpalo and it seems like we might have an intermediate solution that would still use the KES keys, which seems preferable for security reasons, while still alleviating the need to run mithril-signer alongside the block producing node, which seems a key issue for most SPO IIUC. 
Details still need to be figured out but the idea is have a separate process, launched at some defined interval (eg. using `cron` scheduling), that would connect to the mithril-signer server running alongside a relay node, do the signing, and send the signature to the signer which will be responsible for handling registration with the aggregator. This seems to also make sense in the (yet to be defined) fully decentralised setting (eg. no aggregator) whereby the signer.

**Comment by iquerejeta** [comment]
That is a good point. We were thinking of doing exactly what is proposed in CIP 22 but with the KES key instead of the VRF key. I wonder what was the rationale of CIP 22 to choose VRF keys instead of KES keys to produce these signatures. Probably simple to transition to VRF if we choose to go that path. 

I'm wondering about long range attacks. Let's assume that an adversary manages to attack a few SPOs at the same time, at a given point in time. If we register with the VRF keys, then it would be possible to mimic a past registration, create fresh mithril keys to sign a fake mithril chain. If we used KES keys, then an adversary that attacks SPOs at a given point in time would only be able to sign registrations for that epoch, and not previous ones.

**Comment by ghost** [comment]
Is this not superseded by #508 ?

**Reply by iquerejeta** [reply]
If we agree to the design discussed there, yes.

**Reply by ghost** [reply]
🤔 Having 2 discussions about the same topic could be confusing.

**Reply by iquerejeta** [reply]
Sounds good. Let's close this 👍
